Observed a panic while running `terraform apply`

Stack trace from the terraform-provider-honeycombio_v0.23.0 plugin:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0xb8cdd4]

goroutine 105 [running]:
github.com/honeycombio/terraform-provider-honeycombio/client.(*Client).retryHTTPCheck(0x7e43a0?, {0xef49e8, 0xc0003aae70}, 0x0, {0xc0007b4ab8?, 0x0?})
	github.com/honeycombio/terraform-provider-honeycombio/client/client.go:240 +0x34
github.com/hashicorp/go-retryablehttp.(*Client).Do(0xc00045d5e0, 0xc000622438)
	github.com/hashicorp/go-retryablehttp@v0.7.5/client.go:656 +0x66d
github.com/honeycombio/terraform-provider-honeycombio/client.(*Client).Do(0xc0004e4000, {0xef49e8, 0xc0003aae70}, {0xd99912, 0x4}, {0xc0007a4348?, 0x16?}, {0xca5580?, 0xc00054de00?}, {0xca5580, ...})
	github.com/honeycombio/terraform-provider-honeycombio/client/client.go:217 +0x1fb
github.com/honeycombio/terraform-provider-honeycombio/client.(*queries).Create(0xc000426168, {0xef49e8, 0xc0003aae70}, {0xc00062e570, 0xb}, 0xc0002430e0?)
	github.com/honeycombio/terraform-provider-honeycombio/client/query.go:37 +0xf4
github.com/honeycombio/terraform-provider-honeycombio/internal/provider.(*queryResource).Create(0xc00007e5f0, {0xef49e8, 0xc0003aae70}, {{{{0xef95f8, 0xc000242150}, {0xc8fce0, 0xc0002420c0}}, {0xefb7d8, 0xc00013e7d0}}, {{{0xef95f8, ...}, ...}, ...}, ...}, ...)
	github.com/honeycombio/terraform-provider-honeycombio/internal/provider/query_resource.go:112 +0x26e
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).CreateResource(0xc000278ea0, {0xef49e8, 0xc0003aae70}, 0xc0007b54e0, 0xc0007b5480)
	github.com/hashicorp/terraform-plugin-framework@v1.6.1/internal/fwserver/server_createresource.go:101 +0x578
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ApplyResourceChange(0xc0007b5638?, {0xef49e8, 0xc0003aae70}, 0xc00063b310, 0xc0007b5638)
	github.com/hashicorp/terraform-plugin-framework@v1.6.1/internal/fwserver/server_applyresourcechange.go:57 +0x4a5
github.com/hashicorp/terraform-plugin-framework/internal/proto5server.(*Server).ApplyResourceChange(0xc000278ea0, {0xef49e8?, 0xc0003aab40?}, 0xc00063b270)
	github.com/hashicorp/terraform-plugin-framework@v1.6.1/internal/proto5server/server_applyresourcechange.go:55 +0x3e5
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0xef4a20?, {0xef49e8?, 0xc0000b9ce0?}, 0xc00063b270)
	github.com/hashicorp/terraform-plugin-mux@v0.15.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000257680, {0xef49e8?, 0xc0006a1a10?}, 0xc0000ee930)
	github.com/hashicorp/terraform-plugin-go@v0.22.1/tfprotov5/tf5server/server.go:846 +0x3d0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xd73380?, 0xc000257680}, {0xef49e8, 0xc0006a1a10}, 0xc0004eb800, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.22.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:518 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001b9000, {0xef49e8, 0xc0006a1980}, {0xefa460, 0xc000402820}, 0xc0006a6480, 0xc0003aac00, 0x1495578, 0x0)
	google.golang.org/grpc@v1.62.1/server.go:1386 +0xe23
google.golang.org/grpc.(*Server).handleStream(0xc0001b9000, {0xefa460, 0xc000402820}, 0xc0006a6480)
	google.golang.org/grpc@v1.62.1/server.go:1797 +0x100c
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.62.1/server.go:1027 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 20
	google.golang.org/grpc@v1.62.1/server.go:1038 +0x135
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0xb8cdd4]

goroutine 106 [running]:
github.com/honeycombio/terraform-provider-honeycombio/client.(*Client).retryHTTPCheck(0x7e43a0?, {0xef49e8, 0xc0006a4840}, 0x0, {0xc00028cab8?, 0x0?})
	github.com/honeycombio/terraform-provider-honeycombio/client/client.go:240 +0x34
github.com/hashicorp/go-retryablehttp.(*Client).Do(0xc00045d5e0, 0xc00074a078)
	github.com/hashicorp/go-retryablehttp@v0.7.5/client.go:656 +0x66d
github.com/honeycombio/terraform-provider-honeycombio/client.(*Client).Do(0xc0004e4000, {0xef49e8, 0xc0006a4840}, {0xd99912, 0x4}, {0xc00071a108?, 0x16?}, {0xca5580?, 0xc0007be240?}, {0xca5580, ...})
	github.com/honeycombio/terraform-provider-honeycombio/client/client.go:217 +0x1fb
github.com/honeycombio/terraform-provider-honeycombio/client.(*queries).Create(0xc000426168, {0xef49e8, 0xc0006a4840}, {0xc00011a680, 0xb}, 0xc000138120?)
	github.com/honeycombio/terraform-provider-honeycombio/client/query.go:37 +0xf4
github.com/honeycombio/terraform-provider-honeycombio/internal/provider.(*queryResource).Create(0xc000426050, {0xef49e8, 0xc0006a4840}, {{{{0xef95f8, 0xc0006a5a10}, {0xc8fce0, 0xc0006a56b0}}, {0xefb7d8, 0xc00013e7d0}}, {{{0xef95f8, ...}, ...}, ...}, ...}, ...)
	github.com/honeycombio/terraform-provider-honeycombio/internal/provider/query_resource.go:112 +0x26e
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).CreateResource(0xc000278ea0, {0xef49e8, 0xc0006a4840}, 0xc00028d4e0, 0xc00028d480)
	github.com/hashicorp/terraform-plugin-framework@v1.6.1/internal/fwserver/server_createresource.go:101 +0x578
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ApplyResourceChange(0xc00028d638?, {0xef49e8, 0xc0006a4840}, 0xc00013f450, 0xc00028d638)
	github.com/hashicorp/terraform-plugin-framework@v1.6.1/internal/fwserver/server_applyresourcechange.go:57 +0x4a5
github.com/hashicorp/terraform-plugin-framework/internal/proto5server.(*Server).ApplyResourceChange(0xc000278ea0, {0xef49e8?, 0xc0006a4720?}, 0xc00013f270)
	github.com/hashicorp/terraform-plugin-framework@v1.6.1/internal/proto5server/server_applyresourcechange.go:55 +0x3e5
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0xef4a20?, {0xef49e8?, 0xc0006a4420?}, 0xc00013f270)
	github.com/hashicorp/terraform-plugin-mux@v0.15.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000257680, {0xef49e8?, 0xc0001d4450?}, 0xc000228000)
	github.com/hashicorp/terraform-plugin-go@v0.22.1/tfprotov5/tf5server/server.go:846 +0x3d0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xd73380?, 0xc000257680}, {0xef49e8, 0xc0001d4450}, 0xc000550080, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.22.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:518 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001b9000, {0xef49e8, 0xc0001d4390}, {0xefa460, 0xc000402820}, 0xc0006a65a0, 0xc0003aac00, 0x1495578, 0x0)
	google.golang.org/grpc@v1.62.1/server.go:1386 +0xe23
google.golang.org/grpc.(*Server).handleStream(0xc0001b9000, {0xefa460, 0xc000402820}, 0xc0006a65a0)
	google.golang.org/grpc@v1.62.1/server.go:1797 +0x100c
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.62.1/server.go:1027 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 20
	google.golang.org/grpc@v1.62.1/server.go:1038 +0x135
```

Error: The terraform-provider-honeycombio_v0.23.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

--- a/client/client.go
+++ b/client/client.go
@@ -237,8 +237,12 @@ func (c *Client) retryHTTPCheck(ctx context.Context, resp *http.Response, err er
 		return false, ctx.Err()
 	}
 
-	if resp.StatusCode == http.StatusTooManyRequests || resp.StatusCode == http.StatusBadGateway || resp.StatusCode == http.StatusGatewayTimeout {
-		return true, nil
+	if resp != nil {
+		if resp.StatusCode == http.StatusTooManyRequests ||
+			resp.StatusCode == http.StatusBadGateway ||
+			resp.StatusCode == http.StatusGatewayTimeout {
+			return true, nil
+		}
 	}
 
 	return false, nil

